aten/src/ATen/native/Onehot.cpp

```diff
@@ -22,7 +36,9 @@ Tensor one_hot(const Tensor &self, int64_t num_classes) {
             self.key_set().has_all(DispatchKeySet(DispatchKey::Python))) {
         // functional version that torch.compiles better and works with dynamic shapes
         if (num_classes == -1) {
-          num_classes = self.max().item().toLong() + 1;
+            num_classes = self.max().item().toLong() + 1;
+        } else {
+            check_num_classes(self, num_classes);
```

Comment:
@zeshengzong the test failures are probably coming from the fact that there are places where you code now calls `.item()` where we used to not.

Under compile, we shouldn't try to raise the error here if computing the error requires a h2d sync (via the `item()` call). So the first thing you're going to have to do is tweak this condition to only run in the same cases that `item()` was used before.

From looking at the code, I think.... we can always do the `num_classes should be positive` check, but we can only do the `Class values must be smaller than num_classes` check if the `num_classes` field has already been computed